src/App.js

```diff
@@ -1,17 +1,84 @@
 import React from "react";
-import logo from "./logo.png";
 import "./App.css";
+import axios from "axios";
 
 class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      input: " ",
+      weatherData: [],
+    };
+  }
+
+  handleChange = (e) => {
+    const inputValue = e.target.value;
+    this.setState({ input: inputValue });
+  };
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    axios
+      .get(
+        `https://api.openweathermap.org/geo/1.0/direct?q=${this.state.input}&limit=1&appid=${process.env.REACT_APP_API_KEY}`
+      )
+      // City geo data is in response.data[0]
+      // Arrow functions with no curly braces return value after arrow
+      .then((response) => response.data[0])
+      .then((cityGeoData) =>
+        axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?lat=${cityGeoData.lat}&lon=${cityGeoData.lon}&appid=${process.env.REACT_APP_API_KEY}&units=metric`
+        )
+      )
+      .then((response) => {
+        const { data: weatherData } = response;
+        this.setState({
+          weatherData: [...this.state.weatherData, weatherData],
+        });
+      });
+  };
+
   render() {
     return (
       <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-        </header>
+        <h1>Welcome to my weather app 🤠</h1>
+        <p>
+          Please type in any city's name into the input field below and click
+          "submit" to get the weather info 😉
+        </p>
+
+        <form>
+          <input
+            className="input-field"
+            type="text"
+            placeholder="enter the city name"
+            value={this.state.input}
+            onChange={this.handleChange}
+          ></input>
+          <input
+            className="button"
+            type="submit"
+            value="submit"
+            onClick={this.handleSubmit}
+          ></input>
+        </form>
+
+        {this.state.weatherData &&
+          this.state.weatherData.length > 0 &&
+          this.state.weatherData.map((location) => (
```

Comment:
```suggestion
        {this.state.weatherData.map((location) => (
```

This would be sufficient. An array is always truthy, thus `this.state.weatherData` is always true.
When running `.map()` on an empty array, we do not run the callback function, so just running map directly is totally fine.